versions-common/src/main/java/org/codehaus/mojo/versions/ordering/ComparableVersion.java

```diff
@@ -283,6 +287,19 @@ public String toString() {
         }
     }
 
+    /**
+     * Get a ComparableVersion representing the version in a string.
+     */
+    public static ComparableVersion of(String version) {
+        return CACHE.computeIfAbsent(version, ComparableVersion::new);
+    }
+
+    /**
+     * Create a ComparableVersion from a string.
+     *
+     * @deprecated Use {@link #of(String)} instead to use an internal cache.
+     */
+    @Deprecated
     public ComparableVersion(String version) {
```

Comment:
Is there this constructor usage somewhere?
If no I would to consider to make a private it instead deprecating.

Reply:
BoundComparableVersion extends ComparableVersion, so we could make the constructor protected?

Reply:
> BoundComparableVersion extends ComparableVersion, so we could make the constructor protected?

ok

Reply:
Changed to protected.